packages/roosterjs-editor-plugins/lib/plugins/Picker/PickerPlugin.ts

```diff
@@ -53,18 +53,65 @@ const UNIDENTIFIED_CODE = [0, 229];
  */
 export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvider>
     implements EditorPlugin {
-    private editor: IEditor;
-    private eventHandledOnKeyDown: boolean;
-    private blockSuggestions: boolean;
-    private isSuggesting: boolean;
-    private lastKnownRange: Range;
+    private editor: IEditor | undefined = undefined;
+    private core: PickerPluginCore | undefined = undefined;
+    public dataProvider: T | undefined = undefined;
+    private pickerOptions: PickerPluginOptions | undefined = undefined;
+
+    constructor(dataProvider: T, pickerOptions: PickerPluginOptions) {
+        this.dataProvider = dataProvider;
+        this.pickerOptions = pickerOptions;
+    }
+
+    public willHandleEventExclusively(event: PluginEvent) {
+        return this.core ? this.core.willHandleEventExclusively(event) : false;
+    }
+
+    public initialize(editor: IEditor) {
+        this.editor = editor;
+        if (this.editor && this.dataProvider && this.pickerOptions) {
+            this.core = new PickerPluginCore(this.editor, this.dataProvider, this.pickerOptions);
+            this.core.initialize(editor);
+        }
+    }
+
+    public onPluginEvent(event: PluginEvent) {
+        if (this.editor && this.dataProvider && this.core) {
+            this.core.onPluginEvent(event);
+        }
+    }
+
+    public dispose() {
+        this.editor = undefined;
+        this.dataProvider = undefined;
+        this.pickerOptions = undefined;
```

Comment:
Setting them to undefined should be helpful for mem leak, a side effect is that this plugin instance will become unusable because there is no chance to reset them.

This should be ok for now, since we don't expect an instance of plugin to be reused. But not sure if 3rd party people will do this, after all we didn't disable this way of use.

Let's open an issue to track this, and for this fix it is ok here I think.